internal/services/dynatrace/dynatrace_tag_rules_resource_test.go

```diff
@@ -0,0 +1,139 @@
+	"github.com/hashicorp/go-azure-helpers/lang/response"
+	"github.com/hashicorp/go-azure-sdk/resource-manager/dynatrace/2023-04-27/tagrules"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/acceptance/check"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/clients"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type TagRulesResource struct {
+	dynatraceInfo dynatraceInfo
+}
+
+func NewTagRulesResource() TagRulesResource {
+	return TagRulesResource{
+		dynatraceInfo: dynatraceInfo{
+			UserCountry:     os.Getenv("DYNATRACE_USER_COUNTRY"),
+			UserEmail:       os.Getenv("DYNATRACE_USER_EMAIL"),
+			UserFirstName:   os.Getenv("DYNATRACE_USER_FIRST_NAME"),
+			UserLastName:    os.Getenv("DYNATRACE_USER_LAST_NAME"),
+			UserPhoneNumber: os.Getenv("DYNATRACE_USER_PHONE_NUMBER"),
+		},
+	}
+}
+
+func TestAccDynatraceTagRules_basic(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dynatrace_tag_rules", "test")
+	r := NewTagRulesResource()
+	r.preCheck(t)
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+	})
+}
+
+func TestAccDynatraceTagRules_requiresImport(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_dynatrace_tag_rules", "test")
+	r := NewTagRulesResource()
+	r.preCheck(t)
+
+	data.ResourceTest(t, r, []acceptance.TestStep{
+		{
+			Config: r.basic(data),
+			Check: acceptance.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.RequiresImportErrorStep(r.requiresImport),
+	})
+}
+
+func (r TagRulesResource) preCheck(t *testing.T) {
+	if r.dynatraceInfo.UserCountry == "" {
+		t.Skipf("DYNATRACE_USER_COUNTRY must be set for acceptance tests")
+	}
+	if r.dynatraceInfo.UserEmail == "" {
+		t.Skipf("DYNATRACE_USER_EMAIL must be set for acceptance tests")
+	}
+	if r.dynatraceInfo.UserFirstName == "" {
+		t.Skipf("DYNATRACE_USER_FIRST_NAME must be set for acceptance tests")
+	}
+	if r.dynatraceInfo.UserLastName == "" {
+		t.Skipf("DYNATRACE_USER_LAST_NAME must be set for acceptance tests")
+	}
+	if r.dynatraceInfo.UserPhoneNumber == "" {
+		t.Skipf("DYNATRACE_USER_PHONE_NUMBER must be set for acceptance tests")
+	}
+}
+
+func (r TagRulesResource) Exists(ctx context.Context, client *clients.Client, state *pluginsdk.InstanceState) (*bool, error) {
+	id, err := tagrules.ParseTagRuleID(state.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	resp, err := client.Dynatrace.TagRulesClient.Get(ctx, *id)
+	if err != nil {
+		if response.WasNotFound(resp.HttpResponse) {
+			return pointer.To(false), nil
+		}
+		return nil, fmt.Errorf("retrieving %s: %+v", *id, err)
+	}
+	return pointer.To(true), nil
+}
+
+func (r TagRulesResource) basic(data acceptance.TestData) string {
```

Comment:
The basic test is typically creating the resource with the minimal required properties, could we change this test to only the required properties and have a separate `complete` test which includes all the properties?

Reply:
Sure. Will fix.